### PR TITLE
#152106730 Validate Date Entered When Incident Occurred

### PR DESCRIPTION
**Pull request by bmuthoga**
#### What does this PR do?
This PR validates the date entered on Slack, bot-side, of when an incident occurred

#### Description of Task to be completed?
- When a user enters a date, the bot should ask them to repeat if it does not follow the format dd-mm-yy
- When a user enters a date, the bot asks them to repeat if it is a future date

#### How should this be manually tested?
Given that you are signed in to a Slack workspace where the bot has been installed:
- Initiate a conversation with the bot
- Select that you want to report an incident
- The bot will prompt you to enter when the incident occurred
- Type in a date
- If the date is invalid or in the future, the bot will prompt you to enter the date again

#### Any background context you want to add?
This will help prevent invalid dates and typos of when an incident occurred when a user is reporting an incident

#### Screenshots
kapture 2017-12-04 at 12 56 29

#### What are the relevant pivotal tracker stories?
[#152106730](https://www.pivotaltracker.com/n/projects/2117172/stories/152106730)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,84 +1,102 @@
...
+            }
+          );
+
+          break;
+        }
+
+        if (isDateFuture(event.text) === true) {
+          sc.chat.postMessage(
+            event.channel,
+            "Non-existent date provided. Try again.",
```
Comment (andela-ik) [inline index.js]: give a more descriptive message, i.e. You cannot report a future incident or Invalid date entered (dd-mm-yy)

File: modules/future_date_validation.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const isDateFuture = entered_date => {
```
Comment (andela-ik) [inline modules/future_date_validation.js]: I think we should add a folder titled validation inside modules, then have the 2 date validation functions in one file called date_validation. Thoughts?

File: index.js (hunk 1 above)
Comment (bmuthoga) [reply index.js]: Cool

File: modules/future_date_validation.js (hunk 2 above)
Comment (bmuthoga) [reply modules/future_date_validation.js]: Coolio, makes sense

**Pull request merged**